snuba/query/expressions.py

```diff
@@ -130,8 +130,12 @@ class Column(Expression):
     Represent a column in the schema of the dataset.
     """
 
-    column_name: str
     table_name: Optional[str]
+    column_name: str
+    # For nested column, this is the path following the main column name.
+    # For tags.key as an example, tags is the column_name and (value,) would
+    # be the path
+    path: Tuple[str, ...] = tuple()
```

Comment:
Clickhouse technically supports only one level of nesting so this could be a bare string. It seems to me that relying on that which is a coincidence and possibly temporary would not be prudent. Considering the complexity between a bare string and a tuple changes very little I figured out I would just not force us into one level of nesting at most.

Reply:
I think that if the only motivator for this is the ClickHouse nested structure, this should reflect what it supports and not some superset of that.